Export properties with null as a value

fabfischer opened the pull request:
Hi,

I noticed that properties having `null` as a value are not exported. These properties are just missing in an exported JSON file. But I wanted/needed all these properties in my current project - so I created a "feature" for it.

You can trigger this behavior with an option, which is passed through to the jsonSerialize method. It's not the nicest way to control it, but I didn't find another way.  Feedback very welcome.

Output without option:
`{
    "id": 123,
    "foo": "bar"
}`

Output with option:
`{
    "id": 123,
    "foo": "bar",
    "null_property": null
}`

vearutop commented:
Hi, thanks for this PR!

This PR it fails `phpstan` check, because `jsonSerialize()` is defined in [`\JsonSerializable`](https://www.php.net/manual/en/jsonserializable.jsonserialize.php) standard interface and does not accept parameters.

That could be fixed by moving the argument to a hidden instance property like [here](https://github.com/swaggest/php-json-schema/commit/48ab6e460dc36e470e8ea94cd17d3d21b5228ee5).

But I'm not sure that would be the best solution, my concern is that such flag has global impact on all properties (of all nested obejcts) being processed within export session. So such flag will not help in case null should be skipped for some particular property, but exported for all the rest.

Maybe a better solution would be to control nullability on property level, e.g. if property schema declares `"type":["null","number"]` then `null` is exported. Would such solution fit your use case?

Also it is currently possible to express nullability by using magical dynamic phpdoc `@property`, like [here](https://github.com/swaggest/php-json-schema/blob/v0.12.17/tests/src/Helper/User.php#L56).

fabfischer commented:
Hi,
thank you for your response. I like your idea to use the type declaration in the schema. 
I changed my code to check whether a property can be null or not. If a property can be null, it is exported as `null` too.

I squashed all commits to have only one.

vearutop commented:
Tagged `v0.12.20`.
